src/components/CardDemo/card-demo.jsx

```diff
@@ -0,0 +1,199 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import CodeMirror from 'react-codemirror';
+import 'codemirror/mode/javascript/javascript';
+import 'codemirror/lib/codemirror.css';
+import 'codemirror/addon/lint/lint.css';
+
+import Button from 'terra-button';
+import Text from 'terra-text';
+import ErrorView from 'terra-clinical-error-view';
+
+import { storeCardDemoJson } from '../../actions/card-demo-actions';
+import styles from './card-demo.css';
+import { Card } from '../Card/card';
+
+export class CardDemo extends Component {
+  constructor(props) {
+    super(props);
+    try {
+      const checkJson = JSON.parse(this.props.tempUserJson);
+      if (checkJson && typeof checkJson === 'object') {
+        this.state = {
+          displayJSONError: false,
+          errorText: '',
+        };
+      }
+    } catch (e) {
+      this.state = {
+        displayJSONError: true,
+        errorText: 'Cannot parse the JSON',
+      };
+    }
+    this.state = {
```

Comment:
Will this not override the `state` irrespective of what you set in the `try` `catch` block ?